apps/manager_app/onboarding/onboarding.c

```diff
@@ -157,32 +155,35 @@ const flow_step_t *onboarding_get_step(void) {
 }
 
 onboarding_steps_e onboarding_get_last_step(void) {
-  // TODO: Get last_step from flash
-  return last_step;
+  return get_onboarding_step();
 }
 
 void onboarding_set_step_done(const onboarding_steps_e next_step) {
   /* Validate next_step */
-  if (next_step > ONBOARDING_COMPLETE) {
+  if (ONBOARDING_COMPLETE < next_step) {
     return;
   }
 
-  // TODO: Get last_step from flash and update next_step in flash
-  /* Ensure we never go back a step */
-  if ((ONBOARDING_COMPLETE != last_step) && (last_step < next_step)) {
-    last_step = next_step;
+  onboarding_steps_e last_step = onboarding_get_last_step();
+
+  /* Check for DEFAULT_VALUE_IN_FLASH to save the state in a virgin device and
+   * ensure we never go back a step */
+  if ((DEFAULT_VALUE_IN_FLASH == last_step) ||
+      ((ONBOARDING_COMPLETE != last_step) && (last_step < next_step))) {
```

Comment:
can simplify this check.

`(ONBOARDING_COMPLETE != last_step) && (last_step < next_step)` is redundant

Reply:
I re-checked the logic and it does seem to be redundant. Updated the same. Thanks